src/sentry/tasks/reports.py

```diff
@@ -0,0 +1,427 @@
+
+
+@instrumented_task(
+    name='sentry.tasks.reports.prepare_reports',
+    queue='reports.prepare')
+def prepare_reports(*args, **kwargs):
+    timestamp, duration = _fill_default_parameters(*args, **kwargs)
+
+    organization_ids = _get_organization_queryset().values_list('id', flat=True)
+    for organization_id in organization_ids:
+        prepare_organization_report.delay(timestamp, duration, organization_id)
+
+
+@instrumented_task(
+    name='sentry.tasks.reports.prepare_organization_report',
+    queue='reports.prepare')
+def prepare_organization_report(timestamp, duration, organization_id):
+    organization = _get_organization_queryset().get(id=organization_id)
+
+    if not features.has('organizations:reports:prepare', organization):
+        return
+
+    # TODO: Build and store project reports here.
+
+    for user_id in organization.member_set.values_list('id', flat=True):
+        deliver_organization_user_report.delay(
+            timestamp,
+            duration,
+            organization_id,
+            user_id,
+        )
+
+
+def fetch_personal_statistics((start, stop), organization, user):
+    resolved_issue_ids = Activity.objects.filter(
+        project__organization_id=organization.id,
+        user_id=user.id,
+        type__in=(
+            Activity.SET_RESOLVED,
+            Activity.SET_RESOLVED_IN_RELEASE,
+        ),
+        datetime__gte=start,
+        datetime__lt=stop,
+        group__status=GroupStatus.RESOLVED,  # only count if the issue is still resolved
+    ).values_list('group_id', flat=True)
+    return {
+        'resolved': len(resolved_issue_ids),
+        'users': tsdb.get_distinct_counts_union(
+            tsdb.models.users_affected_by_group,
+            resolved_issue_ids,
+            start,
+            stop,
+            60 * 60 * 24,
+        ),
+    }
+
+
+@instrumented_task(
+    name='sentry.tasks.reports.deliver_organization_user_report',
+    queue='reports.deliver')
+def deliver_organization_user_report(timestamp, duration, organization_id, user_id):
+    organization = _get_organization_queryset().get(id=organization_id)
+    user = User.objects.get(id=user_id)
+
+    projects = set()
+    for team in Team.objects.get_for_user(organization, user):
+        projects.update(
+            Project.objects.get_for_user(team, user, _skip_team_check=True),
+        )
+
+    interval = _to_interval(timestamp, duration)
+    start, stop = interval
+
+    def fetch_report(project):
+        # TODO: Fetch reports from storage, rather than building on demand.
+        return prepare_project_report(
+            interval,
+            project,
+        )
+
+    report = reduce(
+        merge_reports,
+        map(
+            fetch_report,
+            projects,
+        ),
+    )
+
+    message = MessageBuilder(
+        subject=u'Sentry Report for {}'.format(organization.name),
+        template='sentry/emails/reports/body.txt',
+        html_template='sentry/emails/reports/body.html',
```

Comment:
Please add `type='organization.report'`, somewhere in these kwargs so we can track send stats.